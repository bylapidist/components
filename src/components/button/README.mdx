---
name: Button
menu: Components
---

import { Playground, Props } from 'docz';
import { Button } from './index';

# Button

<Props of={Button} />

## Basic usage

<Playground>
    <Button text="Click me" />
</Playground>

## Button types

<Playground>
    <PrimaryButton text="Click me" />
    <SecondaryButton text="Click me" />
    <TertiaryButton text="Click me" />
    <DangerButton text="Click me" />
</Playground>

## Button modifiers

<Playground>
    <Button text="Click me" block />
    <Button text="Click me" block loading />
    <Button text="Click me" block outline />
    <Button text="Click me" disabled />
    <Button text="Click me" outline small />
    <Button text="Click me" small loading />
    <Button text="Click me" outline disabled small />
</Playground>
