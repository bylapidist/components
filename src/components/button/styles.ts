import { ColorGroup, getProperty, Styles } from '@lapidist/styles';
import { ButtonProps } from './index';

interface ButtonVariantStyles {
    borderColor: ColorGroup | string;
    backgroundColor: ColorGroup | string;
    paddingY: string;
    fontSize: string;
    textColor: ColorGroup | string;
    hoverBackgroundColor: ColorGroup | string;
    hoverTextColor: ColorGroup | string;
    disabledHoverBackgroundColor: ColorGroup | string;
    disabledTextColor: ColorGroup | string;
    disabledBackgroundColor: ColorGroup | string;
}

const buttonSizing = (small?: boolean) => ({
    paddingY: small ? '1' : '2',
    fontSize: small ? '2' : '3'
});

const darkGrey: ColorGroup = { group: 'grey', shade: 'dark' };

const lightBase: ColorGroup = { group: 'base', shade: 'light' };

const lightestGrey: ColorGroup = { group: 'grey', shade: 'lightest' };

const buttonColors = (dark: string, base: string, ghost?: boolean) => ({
    borderColor: dark,
    hoverBackgroundColor: dark,
    hoverTextColor: lightBase,
    backgroundColor: ghost ? 'transparent' : base,
    textColor: ghost ? darkGrey : lightBase,
    disabledHoverBackgroundColor: ghost ? lightestGrey : dark,
    disabledTextColor: ghost ? darkGrey : lightBase,
    disabledBackgroundColor: ghost ? lightestGrey : dark
});

const buttonVariants = ({
    kind,
    theme,
    small,
    ghost
}: ButtonProps): ButtonVariantStyles => {
    const { dark, base } = getProperty<{
        [K: string]: string;
    }>(theme, 'colors', kind);

    return {
        ...buttonSizing(small),
        ...buttonColors(dark, base, ghost)
    };
};

const buttonBaseStyles: Styles = {
    borderWidth: '1',
    textAlign: 'center',
    margin: '2',
    paddingX: '4',
    borderRadius: '2',
    borderStyle: 'solid'
};

const buttonVariantStyles = ({
    borderColor,
    backgroundColor,
    textColor,
    paddingY,
    fontSize,
    hoverBackgroundColor,
    hoverTextColor,
    disabledHoverBackgroundColor,
    disabledTextColor,
    disabledBackgroundColor
}: ButtonVariantStyles): Styles => ({
    ...buttonBaseStyles,
    borderColor,
    backgroundColor,
    textColor,
    paddingY,
    fontSize,
    pseudo: {
        ':hover': {
            cursor: 'pointer',
            backgroundColor: hoverBackgroundColor,
            textColor: hoverTextColor
        },
        ':hover:disabled': {
            cursor: 'not-allowed',
            backgroundColor: disabledHoverBackgroundColor
        },
        ':disabled': {
            opacity: '0.7',
            textColor: disabledTextColor,
            backgroundColor: disabledBackgroundColor
        }
    }
});

export const buttonStyles = (props: ButtonProps): Styles =>
    buttonVariantStyles(buttonVariants(props));
