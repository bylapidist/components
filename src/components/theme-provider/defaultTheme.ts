export interface ColorGroup {
    lightest: string;
    light: string;
    base: string;
    dark: string;
}

export interface SizingGroup {
    xxxs: string;
    xxs: string;
    xs: string;
    s: string;
    m: string;
    l: string;
    xl: string;
    xxl: string;
    xxxl: string;
}

export interface BreakpointGroup {
    s: string;
    m: string;
    l: string;
}

export interface DefaultTheme {
    colors: {
        greys: ColorGroup;
        blues: ColorGroup;
        yellows: ColorGroup;
        greens: ColorGroup;
        reds: ColorGroup;
    };
    sizing: SizingGroup;
    breakpoints: {
        mobile: BreakpointGroup;
        tablet: BreakpointGroup;
        laptop: BreakpointGroup;
    };
}

export const greys: ColorGroup = {
    lightest: '#ffffff',
    light: '#f3f5f9',
    base: '#66687d',
    dark: '#292b3e'
};

export const blues: ColorGroup = {
    lightest: '#9bfff8',
    light: '#4AD6D8',
    base: '#17a3a5',
    dark: '#007072'
};

export const yellows: ColorGroup = {
    lightest: '#fdffb5',
    light: '#fff08f',
    base: '#ebc764',
    dark: '#c99944'
};

export const greens: ColorGroup = {
    lightest: '#e2f7c9',
    light: '#A7D981',
    base: '#72af5d',
    dark: '#5A8C34'
};

export const reds: ColorGroup = {
    lightest: '#FF848B',
    light: '#E75158',
    base: '#b41e25',
    dark: '#810000'
};

export const defaultTheme: DefaultTheme = {
    colors: {
        greys,
        blues,
        yellows,
        greens,
        reds
    },
    sizing: {
        xxxs: '2px',
        xxs: '4px',
        xs: '10px',
        s: '14px',
        m: '16px',
        l: '24px',
        xl: '36px',
        xxl: '45px',
        xxxl: '57px'
    },
    breakpoints: {
        mobile: {
            s: '320px',
            m: '375px',
            l: '425px'
        },
        tablet: {
            s: '580px',
            m: '640px',
            l: '768px'
        },
        laptop: {
            s: '1024px',
            m: '1280px',
            l: '1440px'
        }
    }
};

export default defaultTheme;
