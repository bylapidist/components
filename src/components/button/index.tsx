import React from 'react';
import deepMerge from 'lodash.merge';
import { Text } from '../text';
import { BoxProps } from '../box';
import { getProperty, Theme } from '@lapidist/styles';
import { withTheme } from 'styled-components';

export type ButtonPropType = BoxProps &
    React.ButtonHTMLAttributes<HTMLButtonElement> &
    React.HTMLProps<HTMLButtonElement>;

export interface ButtonProps {
    readonly kind: string;
    readonly small?: boolean;
    readonly ghost?: boolean;
    readonly theme: Theme;
}

const BaseButton: React.FC<ButtonPropType & ButtonProps> = ({
    as = 'button',
    styles,
    theme,
    small,
    ghost,
    kind,
    ...restProps
}) => {
    const color: { [K: string]: string } = getProperty<{
        [K: string]: string;
    }>(theme, 'colors', kind);

    return (
        <Text
            as={as}
            styles={deepMerge(
                {
                    borderWidth: '1',
                    borderRadius: '2',
                    borderStyle: 'solid',
                    borderColor: color['dark'],
                    backgroundColor: ghost ? 'transparent' : color['base'],
                    textColor: ghost
                        ? { group: 'grey', shade: 'dark' }
                        : { group: 'base', shade: 'light' },
                    textAlign: 'center',
                    paddingX: '2',
                    paddingY: small ? '1' : '2',
                    fontSize: small ? '2' : '3',
                    pseudo: {
                        ':hover': {
                            cursor: 'pointer',
                            backgroundColor: color['dark'],
                            textColor: { group: 'base', shade: 'light' }
                        },
                        ':hover:disabled': {
                            cursor: 'not-allowed',
                            backgroundColor: ghost
                                ? { group: 'grey', shade: 'lightest' }
                                : color['dark']
                        },
                        ':disabled': {
                            opacity: '0.7',
                            textColor: ghost
                                ? { group: 'grey', shade: 'dark' }
                                : { group: 'base', shade: 'light' },
                            backgroundColor: ghost
                                ? { group: 'grey', shade: 'lightest' }
                                : color['dark']
                        }
                    }
                },
                styles
            )}
            {...restProps}
        />
    );
};

export const Button = withTheme(BaseButton);

Button.displayName = 'Button';
