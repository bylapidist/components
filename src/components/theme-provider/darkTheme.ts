import { Theme } from '@lapidist/styles';

export const darkTheme: Theme = {
    boxShadows: {
        '1':
            '0 1px 3px 0 rgba(255, 255, 255, 0.1), 0 1px 2px 0 rgba(255, 255, 255, 0.06)'
    },
    colors: {
        base: {
            light: 'black',
            dark: 'white',
            transparent: 'transparent'
        },
        grey: {
            lightest: '#292B3E',
            light: '#73748b',
            base: '#d4d4d4',
            dark: '#E8E8E8'
        },
        primary: {
            lightest: '#1a8385',
            light: '#17a3a5',
            base: '#4AD6D8',
            dark: '#9bfff8'
        },
        secondary: {
            lightest: '#c99944',
            light: '#ebc764',
            base: '#fff08f',
            dark: '#fdffb5'
        },
        tertiary: {
            lightest: '#5A8C34',
            light: '#72af5d',
            base: '#A7D981',
            dark: '#e2f7c9'
        },
        danger: {
            lightest: '#810000',
            light: '#b41e25',
            base: '#E75158',
            dark: '#FF848B'
        }
    }
};
