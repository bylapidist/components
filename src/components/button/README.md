The Button component provides an API to a consistent button.

```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary">Primary</Button>
        <Button type="button" kind="secondary">Secondary</Button>
        <Button type="button" kind="tertiary">Tertiary</Button>
        <Button type="button" kind="danger">Danger</Button>
        <Button type="button" kind="grey">Grey</Button>
    </div>
</ThemeProvider>
```

Disabled buttons 

```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary" disabled>Primary</Button>
        <Button type="button" kind="secondary" disabled>Secondary</Button>
        <Button type="button" kind="tertiary" disabled>Tertiary</Button>
        <Button type="button" kind="danger" disabled>Danger</Button>
        <Button type="button" kind="grey" disabled>Grey</Button>
    </div>
</ThemeProvider>
```

Small buttons
```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary" small>Primary</Button>
        <Button type="button" kind="secondary" small>Secondary</Button>
        <Button type="button" kind="tertiary" small>Tertiary</Button>
        <Button type="button" kind="danger" small>Danger</Button>
        <Button type="button" kind="grey" small>Grey</Button>
    </div>
</ThemeProvider>
```

Ghost buttons
```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary" ghost>Primary</Button>
        <Button type="button" kind="secondary" ghost>Secondary</Button>
        <Button type="button" kind="tertiary" ghost>Tertiary</Button>
        <Button type="button" kind="danger" ghost>Danger</Button>
        <Button type="button" kind="grey" ghost>Grey</Button>
    </div>
</ThemeProvider>
```

Disabled ghost buttons
```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary" ghost disabled>Primary</Button>
        <Button type="button" kind="secondary" ghost disabled>Secondary</Button>
        <Button type="button" kind="tertiary" ghost disabled>Tertiary</Button>
        <Button type="button" kind="danger" ghost disabled>Danger</Button>
        <Button type="button" kind="grey" ghost disabled>Grey</Button>
    </div>
</ThemeProvider>
```

Small ghost buttons
```jsx harmony
import { ThemeProvider, Button } from '@lapidist/components';

<ThemeProvider>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Button type="button" kind="primary" ghost small>Primary</Button>
        <Button type="button" kind="secondary" ghost small>Secondary</Button>
        <Button type="button" kind="tertiary" ghost small>Tertiary</Button>
        <Button type="button" kind="danger" ghost small>Danger</Button>
        <Button type="button" kind="grey" ghost small>Grey</Button>
    </div>
</ThemeProvider>
```
