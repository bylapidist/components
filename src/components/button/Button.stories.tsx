import * as React from 'react';
import { faXmark } from '@fortawesome/free-solid-svg-icons';
import { ThemeProvider } from '../theme-provider';
import { Button } from './index';

export default {
    title: 'Components/Button',
    component: Button,
    decorators: [(getStory) => <ThemeProvider>{getStory()}</ThemeProvider>]
};

const templateStyles = {};

const DefaultTemplate = (args) => <Button {...args}>Click Me</Button>;

export const Primary = DefaultTemplate.bind({});
Primary.args = {
    kind: 'primary',
    styles: templateStyles
};

export const Secondary = DefaultTemplate.bind({});
Secondary.args = {
    kind: 'secondary',
    styles: templateStyles
};

export const Tertiary = DefaultTemplate.bind({});
Tertiary.args = {
    kind: 'tertiary',
    styles: templateStyles
};

export const Danger = DefaultTemplate.bind({});
Danger.args = {
    kind: 'danger',
    styles: templateStyles
};

export const WithIcon = DefaultTemplate.bind({});
WithIcon.args = {
    kind: 'primary',
    icon: faXmark,
    styles: templateStyles
};

export const AsAnchor = DefaultTemplate.bind({});
AsAnchor.args = {
    as: 'a',
    kind: 'primary',
    href: 'https://google.com',
    target: '_blank',
    styles: templateStyles
};
