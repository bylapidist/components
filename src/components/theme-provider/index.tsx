import React, { Context, PropsWithChildren } from 'react';
import {
    ThemeProvider as StyledThemeProvider,
    createGlobalStyle
} from 'styled-components';
import { mergeThemes, Theme } from '@lapidist/styles';
import { defaultTheme } from './defaultTheme';
import { darkTheme } from './darkTheme';

const GlobalStyle = createGlobalStyle`
    html, body, div, span, applet, object, iframe,
    h1, h2, h3, h4, h5, h6, p, blockquote, pre,
    a, abbr, acronym, address, big, cite, code,
    del, dfn, em, img, ins, kbd, q, s, samp,
    small, strike, strong, sub, sup, tt, var,
    b, u, i, center, button,
    dl, dt, dd, menu, ol, ul, li,
    fieldset, form, label, legend,
    table, caption, tbody, tfoot, thead, tr, th, td,
    article, aside, canvas, details, embed,
    figure, figcaption, footer, header, hgroup,
    main, menu, nav, output, ruby, section, summary,
    time, mark, audio, video {
        margin: 0;
        padding: 0;
        border: 0;
        font-size: 100%;
        font: inherit;
        vertical-align: baseline;
        transition: color 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s,
            border 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s,
            background-color 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s;
    }
    html { box-sizing: border-box; }
    *, *:before, *:after { box-sizing: inherit; }
    article, aside, details, figcaption, figure,
    footer, header, hgroup, main, menu, nav, section { display: block; }
    *[hidden] { display: none; }
    body { line-height: 1; font-size: 62.5%; }
    menu, ol, ul { list-style: none; }
    blockquote, q { quotes: none; }
    blockquote:before, blockquote:after,
    q:before, q:after { content: ''; content: none; }
    table { border-collapse: collapse; border-spacing: 0; }
    img { display: block; max-width: 100%; }
    a, button { text-decoration: none; outline-offset: 2px; outline: transparent solid 2px; }
    a:focus, button:focus { outline: rgb(235, 199, 100) solid 2px; }
`;

export interface ThemeProviderProps extends PropsWithChildren<unknown> {
    readonly theme?: Theme;
}

export const ThemeContext: Context<Theme> =
    React.createContext<Theme>(defaultTheme);

export const useTheme = () => React.useContext(ThemeContext);

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
    children,
    theme
}) => {
    const initialTheme: Theme = useTheme();
    const baseTheme: Theme | undefined = mergeThemes(initialTheme, theme);

    const [isDarkMode, setIsDarkMode] = React.useState<boolean>();
    const [themeContext, setThemeContext] =
        React.useState<Theme | undefined>(baseTheme);

    React.useEffect(() => {
        const isDarkModePersisted =
            localStorage.getItem('isDarkMode') === 'true';
        setIsDarkMode(isDarkModePersisted);
    }, [setIsDarkMode]);

    React.useEffect(() => {
        setThemeContext(
            isDarkMode ? mergeThemes(baseTheme, darkTheme) : baseTheme
        );
    }, [isDarkMode, baseTheme]);

    return (
        <>
            <GlobalStyle />
            <StyledThemeProvider theme={themeContext}>
                {children}
            </StyledThemeProvider>
        </>
    );
};

ThemeProvider.displayName = 'ThemeProvider';
