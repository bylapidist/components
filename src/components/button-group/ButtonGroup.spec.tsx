import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { ThemeProvider } from '../theme-provider';
import { Button } from '../button';
import {
    ButtonGroup,
    ButtonGroupChildrenType,
    ButtonGroupPropType
} from './index';

const sameButtons = [
    <Button key="0" type="button" kind="primary">
        Button
    </Button>,
    <Button key="1" type="button" kind="primary">
        Button
    </Button>,
    <Button key="2" type="button" kind="primary">
        Button
    </Button>
];

const differentButtons = [
    <Button key="0" type="button" kind="primary">
        Button
    </Button>,
    <Button key="1" type="button" kind="secondary">
        Button
    </Button>,
    <Button key="2" type="button" kind="tertiary">
        Button
    </Button>
];

const twoButtons = [
    <Button key="0" type="button" kind="primary">
        Button
    </Button>,
    <Button key="1" type="button" kind="primary">
        Button
    </Button>
];

const setup = (
    props?: ButtonGroupPropType,
    children: ButtonGroupChildrenType = sameButtons
) =>
    render(
        <ThemeProvider>
            <ButtonGroup {...props}>{children}</ButtonGroup>
        </ThemeProvider>
    );

test('it works with same buttons', () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
});

test('it works with different buttons', () => {
    const { container } = setup({}, differentButtons);
    expect(container.firstChild).toMatchSnapshot();
});

test('it works with two buttons', () => {
    const { container } = setup({}, twoButtons);
    expect(container.firstChild).toMatchSnapshot();
});
