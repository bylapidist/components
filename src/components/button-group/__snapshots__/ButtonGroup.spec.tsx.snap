// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works with different buttons 1`] = `
.c0 > :nth-child(n) {
  border-radius: 0;
}

.c0 > :not(:last-child):nth-child(n) {
  margin-right: 0;
}

.c0 > :not(:first-child):nth-child(n) {
  border-left-width: 0;
  margin-left: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  font-size: 1rem;
  font-weight: 400;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  border-radius: 0.25rem;
  border-width: 1px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  margin: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #17a3a5;
  color: white;
  border-color: #1a8385;
}

.c1:hover {
  cursor: pointer;
  background-color: #1a8385;
  color: white;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #1a8385;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #1a8385;
  color: white;
}

.c2 {
  border-style: solid;
  text-align: center;
  font-size: 1rem;
  font-weight: 400;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  border-radius: 0.25rem;
  border-width: 1px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  margin: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #ebc764;
  color: white;
  border-color: #937032;
}

.c2:hover {
  cursor: pointer;
  background-color: #937032;
  color: white;
}

.c2:hover:disabled {
  cursor: not-allowed;
  background-color: #937032;
}

.c2:disabled {
  opacity: 0.7;
  background-color: #937032;
  color: white;
}

.c3 {
  border-style: solid;
  text-align: center;
  font-size: 1rem;
  font-weight: 400;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  border-radius: 0.25rem;
  border-width: 1px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  margin: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #72af5d;
  color: white;
  border-color: #548230;
}

.c3:hover {
  cursor: pointer;
  background-color: #548230;
  color: white;
}

.c3:hover:disabled {
  cursor: not-allowed;
  background-color: #548230;
}

.c3:disabled {
  opacity: 0.7;
  background-color: #548230;
  color: white;
}

<div
  class="c0"
  role="group"
>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
  <button
    class="c2"
    type="button"
  >
    Button
  </button>
  <button
    class="c3"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`it works with same buttons 1`] = `
.c0 > :nth-child(n) {
  border-radius: 0;
}

.c0 > :not(:last-child):nth-child(n) {
  margin-right: 0;
}

.c0 > :not(:first-child):nth-child(n) {
  border-left-width: 0;
  margin-left: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  font-size: 1rem;
  font-weight: 400;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  border-radius: 0.25rem;
  border-width: 1px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  margin: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #17a3a5;
  color: white;
  border-color: #1a8385;
}

.c1:hover {
  cursor: pointer;
  background-color: #1a8385;
  color: white;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #1a8385;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #1a8385;
  color: white;
}

<div
  class="c0"
  role="group"
>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`it works with two buttons 1`] = `
.c0 > :nth-child(n) {
  border-radius: 0;
}

.c0 > :not(:last-child):nth-child(n) {
  margin-right: 0;
}

.c0 > :not(:first-child):nth-child(n) {
  border-left-width: 0;
  margin-left: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  font-size: 1rem;
  font-weight: 400;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  border-radius: 0.25rem;
  border-width: 1px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  margin: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #17a3a5;
  color: white;
  border-color: #1a8385;
}

.c1:hover {
  cursor: pointer;
  background-color: #1a8385;
  color: white;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #1a8385;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #1a8385;
  color: white;
}

<div
  class="c0"
  role="group"
>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
  <button
    class="c1"
    type="button"
  >
    Button
  </button>
</div>
`;
